src/streamlit_extras/star_rating/__init__.py

```diff
@@ -0,0 +1,82 @@
+from htbuilder import span, styles
+from htbuilder.units import px
+from streamlit import streamlit as st
+from streamlit.components.v1 import html
+
+
+def star_rating(rating: float, color="#FFD700"):
+    """
+    Renders a read-only star rating component using htbuilder for Streamlit.
+
+    Args:
+        rating (float): A number in the range [0, 5] representing the rating to be displayed. It can include half values like 2.5.`
```

Comment:
```suggestion
        rating (float): A number in the range [0, 5] representing the rating to be displayed. It can include half values like 2.5.`, and will be rounded to the nearest half-integer.
```